sklearn/metrics/pairwise.py

```diff
@@ -551,7 +565,254 @@ def cosine_distances(X, Y=None):
     return S
 
 
+def gower_distances(X, Y=None, categorical_features=None, apply_scaling=True):
+    """Compute the distances between the observations in X and Y,
+    that may contain mixed types of data, using an implementation
+    of Gower formula.
+
+    Parameters
+    ----------
+    X : array-like, or pandas.DataFrame, shape (n_samples, n_features)
+
+    Y : optional array-like, or pandas.DataFrame, shape (n_samples, n_features)
+
+    categorical_features : optional array-like, shape (n_features)
+        Indicates with True/False whether a column is a categorical attribute.
+        This is useful when categorical atributes are represented as integer
+        values. Categorical ordinal attributes are treated as numeric, and
+        must be marked as false.
+
+        Alternatively, the categorical_features array can be represented only
+        with the numerical indexes of the categorical attribtes.
+
+        If the categorical_features array is not provided, the categorical
+        columns will be auto-detected. Columns with strings, boolean,
+        and other object types will be identified as categorical, and columns
+        with any numeric type will be identified as non categorical.
+
+    apply_scaling : optional boolean, default True
```

Comment:
I think we use `scale` as an input parameter for this semantic. Also, you can have it as:

`scale: boolean, optional (default=True)`, and the same for other parameters.